[Bug] Failing to link when using esp32-s2

### Describe the bug

It appears that the Extern C includes are using C++ based files.
```
.pio/libdeps/lolin_s2_mini/zenoh-pico/src/system/arduino/esp32/network.cpp:18:1: note: 'extern "C"' linkage started here
 extern "C" {
 ^~~~~~~~~~
In file included from /Users/patrick/.platformio/packages/toolchain-xtensa-esp32s2/xtensa-esp32s2-elf/include/c++/8.4.0/algorithm:62,
                 from /Users/patrick/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:184,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform/arduino/esp32.h:18,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform-common.h:36,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform.h:21,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/collections/element.h:21,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/collections/string.h:18,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/src/system/arduino/esp32/network.cpp:25:
/Users/patrick/.platformio/packages/toolchain-xtensa-esp32s2/xtensa-esp32s2-elf/include/c++/8.4.0/bits/stl_algo.h:5696:3: error: template with C linkage
   template<typename _ForwardIterator, typename _Compare>
   ^~~~~~~~
.pio/libdeps/lolin_s2_mini/zenoh-pico/src/system/arduino/esp32/network.cpp:18:1: note: 'extern "C"' linkage started here
 extern "C" {
 ^~~~~~~~~~
In file included from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform/arduino/esp32.h:18,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform-common.h:36,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/system/platform.h:21,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/collections/element.h:21,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/include/zenoh-pico/collections/string.h:18,
                 from .pio/libdeps/lolin_s2_mini/zenoh-pico/src/system/arduino/esp32/network.cpp:25:
/Users/patrick/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:209:10: error: conflicting declaration of C function 'uint16_t makeWord(uint8_t, uint8_t)'
 uint16_t makeWord(uint8_t h, uint8_t l);
          ^~~~~~~~
/Users/patrick/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:208:10: note: previous declaration 'uint16_t makeWord(uint16_t)'
 uint16_t makeWord(uint16_t w);
          ^~~~~~~~
In file included from .pio/libdeps/lolin_s2_mini/zenoh-pico/src/system/arduino/esp32/network.cpp:29:
```

### To reproduce

platformio.ini

```
[env:lolin_s2_mini]
platform = espressif32
board = lolin_s2_mini
board_build.mcu = esp32s2
framework = arduino

build_flags = -std=c++17 -DZENOH_COMPILER_GCC

lib_deps = 
	https://github.com/eclipse-zenoh/zenoh-pico
```

### System info

- Esp32-s2 mini

--- a/include/zenoh-pico/system/platform/arduino/esp32.h
+++ b/include/zenoh-pico/system/platform/arduino/esp32.h
@@ -15,8 +15,6 @@
 #ifndef ZENOH_PICO_SYSTEM_ESP32_TYPES_H
 #define ZENOH_PICO_SYSTEM_ESP32_TYPES_H
 
-#include <Arduino.h>
-
 #include "zenoh-pico/config.h"
 
 #if Z_FEATURE_MULTI_THREAD == 1
@@ -24,7 +22,7 @@
 #endif  // Z_FEATURE_MULTI_THREAD == 1
 
 #if Z_FEATURE_MULTI_THREAD == 1
-typedef TaskHandle_t _z_task_t;
+typedef void *_z_task_t;
 typedef void *z_task_attr_t;  // Not used in ESP32
 typedef pthread_mutex_t _z_mutex_t;
 typedef pthread_cond_t _z_condvar_t;

--- a/src/system/arduino/esp32/system.c
+++ b/src/system/arduino/esp32/system.c
@@ -12,6 +12,7 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
+#include <Arduino.h>
 #include <esp_heap_caps.h>
 #include <stddef.h>
 #include <sys/time.h>

